Mage.Sets/src/mage/cards/k/KjeldoranGuard.java

```diff
@@ -0,0 +1,128 @@
+import mage.abilities.effects.OneShotEffect;
+import mage.abilities.effects.common.SacrificeSourceEffect;
+import mage.abilities.effects.common.continuous.BoostTargetEffect;
+import mage.cards.Card;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.*;
+import mage.filter.common.FilterLandPermanent;
+import mage.game.Game;
+import mage.game.events.GameEvent;
+import mage.target.common.TargetCreaturePermanent;
+import mage.target.targetpointer.FixedTarget;
+
+import java.util.UUID;
+
+/**
+ * @author Alex-Vasile, Susucr
+ */
+public final class KjeldoranGuard extends CardImpl {
+
+    private static final FilterLandPermanent snowLandFiler = new FilterLandPermanent("a snow land");
+
+    static { snowLandFiler.add(SuperType.SNOW.getPredicate()); }
+
+    public KjeldoranGuard(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{1}{W}");
+
+        this.subtype.add(SubType.HUMAN);
+        this.subtype.add(SubType.SOLDIER);
+        this.power = new MageInt(1);
+        this.toughness = new MageInt(1);
+
+        // {T}: Target creature gets +1/+1 until end of turn. When that creature leaves the battlefield this turn, sacrifice Kjeldoran Guard. Activate only during combat and only if defending player controls no snow lands.
+        Ability ability = new ConditionalActivatedAbility(
+                Zone.BATTLEFIELD,
+                new KjeldoranGuardEffect(),
+                new TapSourceCost(),
+                new DefendingPlayerControlsNoSourceCondition(snowLandFiler),
+                "activate only during combat and only if defending player controls no snow lands"
+        );
+        ability.addTarget(new TargetCreaturePermanent());
+        this.addAbility(ability);
+    }
+
+    private KjeldoranGuard(final KjeldoranGuard card) { super(card); }
+
+    @Override
+    public Card copy() {
+        return new KjeldoranGuard(this);
+    }
+}
+
+class KjeldoranGuardEffect extends OneShotEffect {
+
+    KjeldoranGuardEffect() {
+        super(Outcome.Neutral);
+        staticText = "Target creature gets +1/+1 until end of turn." +
+                "When that creature leaves the battlefield this turn, sacrifice {this}." +
+                "Activate only during combat and only if defending player controls no snow lands.";
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        UUID targetId = source.getFirstTarget();
+        if (game.getPermanent(targetId) == null) {
+            return false;
+        }
+
+        // Target creature gets +1/+1 until end of turn.
+        BoostTargetEffect buffEffect = new BoostTargetEffect(1, 1, Duration.EndOfTurn);
+        buffEffect.setTargetPointer(new FixedTarget(targetId, game));
+        game.addEffect(buffEffect, source);
+
+        // When that creature leaves the battlefield this turn, sacrifice Kjeldoran Guard.
+        game.addDelayedTriggeredAbility(
```

Comment:
Please test- if you flicker Kjeldoran Guard, shouldn't have to sacrifice it.

Reply:
oh right that should be using MOR

Reply:
Actually that was already correct. `SacrificeSourceEffect` does sacrifice `source.getSourceObjectIfItStillExists(game)`, so the Kjeldoran Guard with a different zcc will not get sacrificed (there still will be a trigger, although it does nothing).

I did replace the UUID of the target to a MOR, but that should not really matter with
- target being still valid means that it has not leave on resolve
- the `DelayedTrigger` being set with `triggerOnce=true`.

Reply:
Oh nevermind, there is indeed an issue if you blink the Guard in response to tapping it.
Will rework with a MOR based SacrificeTargetEffect